### PR TITLE
Add `<|>` to the list of parsable but unused operators

### PR DESCRIPTION
**Pull request by zachallaun**

### CODE REVIEW

File: lib/elixir/pages/references/operators.md [hunk 1]
```diff
@@ -6,28 +6,28 @@ This document is a complete reference of operators in Elixir, how they are parse
...
+---------------------------------------------------- | -------------
+`@`                                                  | Unary
+`.`                                                  | Left
+`+` `-` `!` `^` `not`                                | Unary
+`**`                                                 | Left
+`*` `/`                                              | Left
+`+` `-`                                              | Left
+`++` `--` `+++` `---` `..` `<>`                      | Right
+`in` `not in`                                        | Left
+`\|>` `<<<` `>>>` `<<~` `~>>` `<~` `~>` `<~>` `<\|>` | Left
```
Comment (josevalim) [inline lib/elixir/pages/references/operators.md:19]:
```suggestion
`<\|>` `\|>` `<<<` `>>>` `<<~` `~>>` `<~` `~>` `<~>` | Left
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by sabiwara** [comment]
Should we also remove the deprecation notice introduced in [1.14](https://hexdocs.pm/elixir/1.14.0/changelog.html#4-hard-deprecations)?

```elixir
warning: <|> is deprecated. Use another pipe-like operator

**Comment by zachallaun** [comment]
Ah -- if it's indeed deprecated, my inclination is to add a note to the operator reference. I don't think it needs to be un-deprecated, but I do think that reference documentation should be as complete as possible.

**Comment by josevalim** [comment]
Ah, it was not listed because it was deprecated. We don't want users to add it only to find out it is deprecated, so I will revert this. :) I can see how it is useful if someone is building a parser reference, but in this case, it is easy to left them all from the parser (with precedence info and more).

**Comment by josevalim** [comment]
Although the reason we deprecated `<|>` is because we were considering introducing `<+>`, `<->`, etc as additional numeric operators. This could be used by libraries like decimal (and Nx) to provide an alternative set of mathematical functions that know about "other types". In this case, `<|>` was removed to avoid ambiguity with it being an alternative version of `|`. The plan is now halted, so in theory we could bring `<|>` back, but perhaps best to leave it as is in case we do need to rearrange precedences in the future.

Something else that I have eternally postponed is introducing a `<|` operator, which would nicely fit our existing table, but I am sure someone will use it as pipe right to left and that would be very confusing! :)

**Pull request merged**